### PR TITLE
feat(rpc): Added quorum listextended RPC

### PR DESCRIPTION
**Pull request by ogabrielides**
## Issue being fixed or feature implemented

Introduction of RPC `quorum listextended`.
This change was requested by Platform since the already existing RPC quorum list doesn't return all required data.
`quorum listextended` behaves exactly like `quorum list`, but with a more enriched reply.
Data returned per quorum:
- `quorumHash`
- `creationHeight`
- `quorumIndex`
- `minedBlockHash`

Note: I chose to create a new one, since the reply models differs a lot.
array of strings for `quorum list` compared to array of objects for `quorum listextended`.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: src/rpc/quorums.cpp [hunk 1]
```diff
@@ -82,6 +82,73 @@ static UniValue quorum_list(const JSONRPCRequest& request)
     return ret;
 }
 
+static void quorum_list_extended_help(const JSONRPCRequest& request)
+{
+    RPCHelpMan{"quorum listextended",
+               "Extended list of on-chain quorums\n",
```
Comment (thephez) [inline src/rpc/quorums.cpp:88]: The indentation in here looks wider than usual and a bit inconsistent.

**Review by thephez — CHANGES_REQUESTED** [review]
Been waiting for this :heart: Formatting looks a little off in some places...

**Comment by PastaPastaPasta** [comment]
Release notes need to be added

**Comment by PastaPastaPasta** [comment]
Please implement some functional tests for this. I would expect that to be relatively easy

**Comment by ogabrielides** [comment]
> Please implement some functional tests for this. I would expect that to be relatively easy

@PastaPastaPasta Done

**Review by thephez — APPROVED** [review]
Slightly tested ACK :+1:

**Comment by ogabrielides** [comment]
> Would it not make more sense for this to be a map from quorumHash -> data instead of a list of data blobs?

It's the same after all

**Comment by ogabrielides** [comment]
> Details

@PastaPastaPasta Done

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Review by thephez — DISMISSED** [review]
Slightly test re-ACK

**Review by thephez — COMMENTED** [review]
Actually it looks like the help doesn't return this line as I'd expect:
``` {RPCResult::Type::OBJ, "xxxx", "Quorum hash. Note: most recent quorums come first.",```

This is what I see...
```Result:
{                                  (json object)
  "quorumName" : [                 (json array) List of quorum details per some quorum type
    {                              (json object) Quorum hash. Note: most recent quorums come first.
      "creationHeight" : n,        (numeric) Block height where the DKG started.
      "quorumIndex" : n,           (numeric) Quorum index (applicable only to rotated quorums).
      "minedBlockHash" : "hex"     (string) Blockhash where the commitment was mined.
    },
    ...
  ]
}```

**Review by PastaPastaPasta — APPROVED** [review]
re-ACK

**Review by thephez — APPROVED** [review]
Let's get this merged! :+1:

**Comment by ogabrielides** [comment]
@PastaPastaPasta Yup, ACK

**Pull request merged**